plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/labeling/helper/StyledTextCalculationHelper.xtend

```diff
@@ -96,45 +95,54 @@ class StyledTextCalculationHelper {
 	def dispatch StyledString dispatchGetStyledText(EObjectWithContext objectWithContext) {
 		var StyledString styledText = dispatchGetStyledText(objectWithContext.obj);
 
-		if (objectWithContext.obj instanceof N4MemberDeclaration) {
-			val TMember member = (objectWithContext.obj as N4MemberDeclaration).definedTypeElement;
-			if (member !== null && member.containingType !== null &&
-				member.containingType != objectWithContext.context) {
+		val TMember member = if (objectWithContext.obj instanceof N4MemberDeclaration) {
+				(objectWithContext.obj as N4MemberDeclaration).definedTypeElement
+			} else if (objectWithContext.obj instanceof TMember) {
+				objectWithContext.obj as TMember
+			} else
+				null;
 
-				val orgDest = " from " + labelCalculationHelper.dispatchDoGetText(member.containingType);
+		if (member !== null && member.containingType !== null && member.containingType != objectWithContext.context) {
 
-				if (member.polyfilled) {
-					styledText.append(" polyfilled" + orgDest, N4JSStylers.POLYFILLED_MEMBERS_STYLER);
-				} else if (member.containingType instanceof TInterface && objectWithContext.context instanceof TClass) {
-					styledText.append(" consumed" + orgDest, N4JSStylers.CONSUMED_MEMBERS_STYLER);
-				} else {
-					styledText.append(" inherited" + orgDest, N4JSStylers.INHERITED_MEMBERS_STYLER);
-				}
+			val orgDest = " from " + labelCalculationHelper.dispatchDoGetText(member.containingType);
 
+			if (member.polyfilled) {
+				styledText.append(" polyfilled" + orgDest, N4JSStylers.POLYFILLED_MEMBERS_STYLER);
+			} else if (member.containingType instanceof TInterface && objectWithContext.context instanceof TClass) {
```

Comment:
is this condition for _consumed member_ correct? The member must have a default implementation in the interface and is not overriden in the class right?